src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -6730,6 +6750,126 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
 
     if ( ! relationFunction || ( geometry.*relationFunction )( feat2.geometry() ) ) // Calls the method provided as template argument for the function (e.g. QgsGeometry::intersects)
     {
+
+      if ( isIntersectsFunc && ( minOverlap != -1 || minInscribedCircleRadius != -1 ) )
+      {
+        const QgsGeometry intersection { geometry.intersection( feat2.geometry() ) };
+
+        switch ( intersection.type() )
+        {
+          case QgsWkbTypes::GeometryType::PolygonGeometry:
+          {
+            // Check min overlap for intersection (if set)
+            if ( intersection.isMultipart() )
+            {
+              bool testResult { false };
+              for ( auto it = intersection.const_parts_begin(); ! testResult && it != intersection.const_parts_end(); ++it )
+              {
+                const QgsPolygon *geom = qgsgeometry_cast< const QgsPolygon * >( *it );
+                // qDebug() << "Area" << feat2.id() << geom->area();
+                if ( minOverlap != -1 )
+                {
+                  if ( geom->area() >= minOverlap )
+                  {
+                    testResult = true;
+                  }
+                  else
+                  {
+                    continue;
+                  }
+                }
+
+                // Check min inscribed circle radius for intersection (if set)
+                if ( minInscribedCircleRadius != -1 )
+                {
+                  const QgsRectangle bbox = geom->boundingBox();
+                  const double width = bbox.width();
+                  const double height = bbox.height();
+                  const double size = width > height ? width : height;
+                  const double tolerance = size / 1000.0;
+                  //qDebug() << "Inscribed circle radius" << feat2.id() << QgsGeos( geom ).maximumInscribedCircle( tolerance )->length();
+                  testResult = QgsGeos( geom ).maximumInscribedCircle( tolerance )->length() >= minInscribedCircleRadius;
```

Comment:
This will raise an exception on geos < 3.9. We'll need to handle that gracefully by raising an evaluation error on the expression instead. (Also document this requirement in the expression help)

Reply:
Still without mentioning GEOS?

Reply:
Yes, I think for that error it's fine (since it's the direct cause of the error, it goes from being "extraneous technical details" to "this is the actual cause of the error")